frontend/src/components/Gallery.js

```diff
@@ -11,6 +11,10 @@ export default function Gallery() {
             .catch(error => console.log(error))
     }, [])
 
+    if (!products) {
+        return <div>Products are loading...</div>
+    }
+
```

Comment:
Maybe you could display a different message when the fetch fails, so your app won't tell the user it's loading even though it has already failed to fetch any data. You could do so by using states (isLoading and isError for example which are set accordingly) and conditional rendering.